ssh/tailssh/tailssh.go

```diff
@@ -255,35 +259,59 @@ func (c *conn) vlogf(format string, args ...any) {
 }
 
 // errDenied is returned by auth callbacks when a connection is denied by the
-// policy. It returns a gossh.BannerError to make sure the message gets
-// displayed as an auth banner.
-func errDenied(message string) error {
+// policy. It writes the message to an auth banner and then returns an empty
+// gossh.PartialSuccessError in order to stop processing authentication
+// attempts and immediately disconnect the client.
+func (c *conn) errDenied(message string) *gossh.PartialSuccessError {
```

Comment:
kinda weird that this is "errFoo" but below you have methods that are "fooError"

Also weird that you're returning a concrete error type .... this is the classic mistake that's documented at https://go.dev/doc/faq#nil_error but I guess you're always returning a non-nil error at least, so probably okay.

Reply:
Yeah, I had made a deliberate choice to return the concrete error type to ensure that no one accidentally changes this to return a different error type, because that would break the invariant that this error should result in authentication terminating. But since I added a check in the defer() in `clientAuth()`, I don't need this anymore.

Reply:
Done.